src/node.cc

```diff
@@ -4531,6 +4531,8 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
     } while (more == true);
   }
 
+  AsyncWrap::RunDestroyCbs(&env);
```

Comment:
> This is why I think we should treat the `destroy` queue exactly like the `setImmediate` queue, by using `uv_check_start`.

You mean you'd prefer we use `uv_check_t` instead of a `uv_idle_` as it's done in `PushBackDestroyId()` (`src/async-wrap.cc`)?

Honestly if you want the callbacks to run before the poll phase then it might be worth it to run it as a `uv_timer_t` with a timeout of `0`. This will cause them to be run before any `setImmediate()`'s are called.